tracer/test/Datadog.Trace.Tests/Agent/MessagePack/SpanMessagePackFormatterTests.cs

```diff
@@ -208,6 +208,230 @@ public void SpanLink_Tag_Serialization()
+
+            attributes2["string_key"].Type.Should().Be(0);
+            attributes2["string_key"].StringValue.Should().Be("hello");
+            attributes2["string_key"].BoolValue.Should().BeNull();
+            attributes2["string_key"].IntValue.Should().BeNull();
+            attributes2["string_key"].DoubleValue.Should().BeNull();
+
+            var boolArray = attributes2["bool_array"];
+            boolArray.Type.Should().Be(4);
+            boolArray.ArrayValue.Values.Should().HaveCount(3);
+            boolArray.ArrayValue.Values.Should().AllSatisfy(item => item.Type.Should().Be(1)); // bool type
+            boolArray.ArrayValue.Values[0].BoolValue.Should().Be(true);
+            boolArray.ArrayValue.Values[1].BoolValue.Should().Be(false);
+            boolArray.ArrayValue.Values[2].BoolValue.Should().Be(true);
+            boolArray.StringValue.Should().BeNull();
+            boolArray.BoolValue.Should().BeNull();
+            boolArray.IntValue.Should().BeNull();
+            boolArray.DoubleValue.Should().BeNull();
+
+            var intArray = attributes2["int_array"];
+            intArray.Type.Should().Be(4);
+            intArray.ArrayValue.Values.Should().HaveCount(3);
+            intArray.ArrayValue.Values.Should().AllSatisfy(item => item.Type.Should().Be(2)); // int type
+            intArray.ArrayValue.Values[0].IntValue.Should().Be(123);
+            intArray.ArrayValue.Values[1].IntValue.Should().Be(1234);
+            intArray.ArrayValue.Values[2].IntValue.Should().Be(12345);
+            intArray.StringValue.Should().BeNull();
+            intArray.BoolValue.Should().BeNull();
+            intArray.IntValue.Should().BeNull();
+            intArray.DoubleValue.Should().BeNull();
+
+            var doubleArray = attributes2["double_array"];
+            doubleArray.Type.Should().Be(4);
+            doubleArray.ArrayValue.Values.Should().HaveCount(3);
+            doubleArray.ArrayValue.Values.Should().AllSatisfy(item => item.Type.Should().Be(3)); // double type
+            doubleArray.ArrayValue.Values[0].DoubleValue.Should().Be(1.2);
+            doubleArray.ArrayValue.Values[1].DoubleValue.Should().Be(1.321);
+            doubleArray.ArrayValue.Values[2].DoubleValue.Should().Be(200000.1);
+            doubleArray.StringValue.Should().BeNull();
+            doubleArray.BoolValue.Should().BeNull();
+            doubleArray.IntValue.Should().BeNull();
+            doubleArray.DoubleValue.Should().BeNull();
+        }
+        else
+        {
+            deserializedSpan.SpanEvents.Should().BeNullOrEmpty();
+            deserializedSpan.Tags.Should().ContainKey("events");
+            var eventsJson = deserializedSpan.Tags["events"];
+
+            eventsJson.Should().Contain($"\"name\":\"{eventName}\"");
+            eventsJson.Should().Contain($"\"time_unix_nano\":{eventTimestamp.ToUnixTimeNanoseconds()}");
+
+            var firstEventExpectedAttributes = new[]
```

Comment:
Can you also assert the values for all of the attributes like byte_key, short_key, etc.?